spring-cloud-netflix-eureka-server/src/test/java/org/springframework/cloud/netflix/eureka/server/EurekaInstanceMonitorTests.java

```diff
@@ -0,0 +1,166 @@
+		instanceRegistry.register(fooInstanceInfo2, false);
+		instanceRegistry.register(barInstanceInfo, false);
+
+		final Map<Tags, Long> counts = Map.of(tags(fooInstanceInfo), 2L, tags(barInstanceInfo), 1L);
+		assertEurekaInstance(counts);
+	}
+
+	@Test
+	void testPartialDeregistrationAfterMultipleRegistrations() {
+		instanceRegistry.register(fooInstanceInfo, false);
+		instanceRegistry.register(fooInstanceInfo2, false);
+		instanceRegistry.register(barInstanceInfo, false);
+		instanceRegistry.register(barInstanceInfo2, false);
+
+		instanceRegistry.internalCancel(fooInstanceInfo.getAppName(), fooInstanceInfo.getInstanceId(), false);
+
+		final Map<Tags, Long> counts = Map.of(tags(fooInstanceInfo), 1L, tags(barInstanceInfo), 2L);
+		assertEurekaInstance(counts);
+	}
+
+	@Test
+	void testPartialDeregistrationAndThenRegistrationAfterMultipleRegistrations() {
+		instanceRegistry.register(fooInstanceInfo, false);
+		instanceRegistry.register(fooInstanceInfo2, false);
+		instanceRegistry.register(barInstanceInfo, false);
+		instanceRegistry.register(barInstanceInfo2, false);
+
+		instanceRegistry.internalCancel(fooInstanceInfo.getAppName(), fooInstanceInfo.getInstanceId(), false);
+		instanceRegistry.register(fooInstanceInfo, false);
+
+		final Map<Tags, Long> counts = Map.of(tags(fooInstanceInfo), 2L, tags(barInstanceInfo), 2L);
+		assertEurekaInstance(counts);
+	}
+
+	@Test
+	void testPartialNonRenewalAfterMultipleRegistrations() {
+		instanceRegistry.register(fooInstanceInfo, false);
+		instanceRegistry.register(fooInstanceInfo2, false);
+		instanceRegistry.register(barInstanceInfo, false);
+		instanceRegistry.register(barInstanceInfo2, false);
+
+		instanceRegistry.statusUpdate(fooInstanceInfo.getAppName(), fooInstanceInfo.getInstanceId(),
+				InstanceInfo.InstanceStatus.DOWN, null, false);
+
+		final Map<Tags, Long> meterRegistryCounts = Map.of(tags(fooInstanceInfo2), 2L);
+		assertEurekaInstance(meterRegistryCounts);
+	}
+
+	@Test
+	void testPartialRenewalAfterMultipleRegistrations() {
+		instanceRegistry.register(fooInstanceInfo, false);
+		instanceRegistry.register(fooInstanceInfo2, false);
+		instanceRegistry.register(barInstanceInfo, false);
+		instanceRegistry.register(barInstanceInfo2, false);
+
+		instanceRegistry.statusUpdate(fooInstanceInfo.getAppName(), fooInstanceInfo.getInstanceId(),
+				InstanceInfo.InstanceStatus.DOWN, null, false);
+		instanceRegistry.statusUpdate(barInstanceInfo2.getAppName(), barInstanceInfo2.getInstanceId(),
+				InstanceInfo.InstanceStatus.STARTING, null, false);
+		instanceRegistry.renew(fooInstanceInfo.getAppName(), fooInstanceInfo.getInstanceId(), false);
+		instanceRegistry.renew(barInstanceInfo2.getAppName(), barInstanceInfo2.getInstanceId(), false);
+
+		final Map<Tags, Long> counts = Map.of(tags(fooInstanceInfo), 1L, tags(fooInstanceInfo2), 1L,
+				tags(barInstanceInfo), 1L, tags(barInstanceInfo2), 1L);
+		assertEurekaInstance(counts);
+	}
+
+	private static Tags tags(InstanceInfo instanceInfo) {
```

Comment:
How hard would it be adding an extra test with a custom, user provided mapping that adds a tag that makes every instance unique (e.g.: `port`)?

Reply:
I will write additional tests 🙏

Reply:
@heowc  Waiting for this one still. Once you've added it, we can go ahead with the merge.

Reply:
Added 😄